lib/active_triples/relation.rb

```diff
@@ -283,19 +281,19 @@ def value_to_node(val)
         valid_datatype?(val) ? RDF::Literal(val) : val
       end
 
-      def add_child_node(resource, object = nil)
+      def add_child_node(resource)
         parent.insert [rdf_subject, predicate, resource.rdf_subject]
 
+        resource = resource.dup
         unless resource.frozen? || 
```

Comment:
Looks like not. Also looks like this is stale code otherwise, since `resource` is also not ever `#frozen?` before the dup call.

I went ahead and removed the conditional.

I'm squashing, then this is ready.